dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/jobService/GetJob.kt

```diff
@@ -80,11 +83,15 @@ class GetJob(private val getJobData: JobData) : JobService(), PrepStage<JobData>
         getJobData.removeJob()
     }
 
-    override fun finishedCompletable(): Completable {
+    override fun finishedCompletable(cancelled: Supplier<Boolean>): Completable {
         return Completable.fromAction {
             val prep = prepare(getJobData)
-            transfer(prep)
-            tearDown()
+            if (!cancelled.get()) {
```

Comment:
To me this would read better if the cancel logic returned instead.  I believe you can do this with kotlin and the `return@fromAction` syntax.